### PR TITLE
feat: resolve-detail flag for get extensions list

### PR DESCRIPTION
**Pull request by MIA-Deltat1995**
## What this PR is for?

Fixed and extended the get extensions by list command to accept the resolve-details flag and support the optional resolution of extension menu and category

### CODE REVIEW

File: internal/clioptions/clioptions.go [hunk 1]
```diff
@@ -85,6 +85,8 @@ type CLIOptions struct {
 	ShowUsers           bool
 	ShowGroups          bool
 	ShowServiceAccounts bool
+
+	ResolveExtensionsDetails bool
```
[the remark this answers is not in the document]
Comment (fredmaggiowski) [reply internal/clioptions/clioptions.go:89]: No, at the moment flags are all coupled and hardcoded here.

@JGiola we should find a way to dynamically register flags with some generic/interface implementation. I might give it a try in the next days

File: internal/cmd/extensions/list.go [hunk 2]
```diff
@@ -43,53 +43,68 @@ func ListCmd(options *clioptions.CLIOptions) *cobra.Command {
...
 	}
 	p.Print()
 }
 
-func menucolumn(extension *extensibility.ExtensionInfo) string {
-	if extension.Menu.ID == "" {
-		return ""
+func menucolumn(extension *extensibility.ExtensionInfo) []string {
+	if extension.Menu == nil {
+		return []string{"", ""}
```
Comment (fredmaggiowski) [inline internal/cmd/extensions/list.go:88]: why not nil, instead?

File: internal/cmd/extensions/list.go [hunk 3]
```diff
@@ -43,53 +43,68 @@ func ListCmd(options *clioptions.CLIOptions) *cobra.Command {
...
+		}
+		p.Record(tableRow...)
 	}
 	p.Print()
 }
 
-func menucolumn(extension *extensibility.ExtensionInfo) string {
-	if extension.Menu.ID == "" {
-		return ""
+func menucolumn(extension *extensibility.ExtensionInfo) []string {
```
Comment (fredmaggiowski) [inline internal/cmd/extensions/list.go:86]: this function is doing the job of two function invocations. I'd simplify the function to manage a single column and invoke it twice

**Review by fredmaggiowski — APPROVED** [review]
LGTM

**Review by JGiola — APPROVED** [review]

**Pull request merged**